include/ignition/rendering/ParticleEmitter.hh

```diff
@@ -36,7 +55,173 @@ namespace ignition
       public virtual Visual
     {
       /// \brief Destructor
-      public: virtual ~ParticleEmitter();
+      public: virtual ~ParticleEmitter() {}
+
+      /// \brief \brief Get the emitter type.
+      /// \return Emitter type.
+      /// \sa EmitterType.
+      /// \sa SetType.
+      public: virtual EmitterType Type() const = 0;
+
+      /// \brief \brief Set the emitter type.
+      /// Default value is EM_POINT.
+      /// \param[in] _type Emitter type.
+      /// \sa EmitterType.
+      /// \sa Type.
+      public: virtual void SetType(const EmitterType _type) = 0;
+
+      /// \brief Get the size of the emitter where the particles are sampled.
+      /// \return The emitter size. See SetEmitterSize() for the interpretation
+      /// of the dimensions depending on the emitter type.
+      /// \sa SetEmitterSize
+      public: virtual ignition::math::Vector3d EmitterSize() const = 0;
+
+      /// \brief Set the size of the emitter where the particles are sampled.
+      /// Default value is (1, 1, 1).
+      /// Note that the interpretation of the emitter area varies depending on
+      /// the emmiter type:
+      ///   - EM_POINT: The area is ignored.
+      ///   - EM_BOX: The area is interpreted as width X height X depth.
+      ///   - EM_CYLINDER: The area is interpreted as the bounding box of the
+      ///                  cilinder. The cylinder is oriented along the Z-axis.
+      ///   - EM_ELLIPSOID: The area is interpreted as the bounding box of an
+      ///                   ellipsoid shaped area, i.e. a sphere or
+      ///                   squashed-sphere area. The parameters are again
+      ///                   identical to EM_BOX, except that the dimensions
+      ///                   describe the widest points along each of the axes.
+      /// \param[in] _size Size of the emitter (width, height, depth).
+      /// \sa EmitterSize
+      public: virtual void SetEmitterSize(
+                  const ignition::math::Vector3d &_size) = 0;
+
+      /// \brief Get how many particles per second should be emitted.
+      /// \return Particles per second.
+      /// \sa SetRate
+      public: virtual double Rate() const = 0;
+
+      /// \brief Set how many particles per second should be emitted.
+      /// Default value is 10.
+      /// \param[in] _rate Particles per second.
+      /// \sa Rate
+      public: virtual void SetRate(double _rate) = 0;
+
+      /// \brief Get the position of the emitter relative to the SceneNode the
+      /// particle system is attached to.
+      /// \return Pose of the emitter.
+      /// \sa SetLocalPose
+      public: virtual ignition::math::Pose3d LocalPose() const = 0;
+
+      /// \brief Set the position of the emitter relative to the SceneNode the
+      /// particle system is attached to.
+      /// Default value is (0, 0, 0, 0, 0, 0).
+      /// \param[in] _pose Pose of the emitter.
+      /// \sa LocalPose
+      public: virtual void SetLocalPose(
```

Comment:
just realized that you shouldn't need to declare `SetLocalPose` and `LocalPose` here because `ParticleEmitter` should inherit [these two functions](https://github.com/ignitionrobotics/ign-rendering/blob/master/include/ignition/rendering/Node.hh#L61) from `Node` already.

Reply:
Good to know. 1843624.